### PR TITLE
Improvements to GL buffer support.

### PR DESCRIPTION
**Pull request by bythos14**
Added glBufferSubData implementation.

Added size variable to gpubuffer struct.

Adjusted and added error checks in glBufferData to be more conformant to the specification.

Ran clang-format on vitaGL.c

### CODE REVIEW

File: source/vitaGL.c [hunk 1]
```diff
@@ -841,17 +842,54 @@ void glDeleteBuffers(GLsizei n, const GLuint *gl_buffers) {
...
 #ifndef SKIP_ERROR_HANDLING
 	if (size < 0) {
 		SET_GL_ERROR(GL_INVALID_VALUE)
 	}
+
+	if (idx == 0) {
+		SET_GL_ERROR(GL_INVALID_OPERATION)
+	}
+#endif
+	vglMemType type = VGL_MEM_VRAM;
```
Comment (Rinnegatamante) [inline source/vitaGL.c:873]: Can you change this so that it uses VRAM only when glUseVram is set to true?

Comment (bythos14) [reply source/vitaGL.c]: Changes made.

**Comment by bythos14** [comment]
Quick question also, [this line](https://github.com/Rinnegatamante/vitaGL/blob/master/source/vitaGL.c#L812) here, specifically the condition `buffer != 0x0000`, does this make it so the default buffer target(0) is -A000 internally?

**Comment by Rinnegatamante** [comment]
To be honest, i can't recall why i've added that check. The buffer impl was pretty rushed and never extensively tested.

**Pull request merged**